components/producers/checkov/main.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"flag"
+	"log"
+
+	"github.com/go-errors/errors"
+
+	v1 "github.com/ocurity/dracon/api/proto/v1"
+	"github.com/ocurity/dracon/pkg/cyclonedx"
+	"github.com/ocurity/dracon/pkg/sarif"
+
+	"github.com/ocurity/dracon/components/producers"
+)
+
+// the CycloneDX target override
+var Target string
+
+func main() {
+	flag.StringVar(&Target, "target", "", "The target being scanned, this will override the CycloneDX target and is useful for cases where you scan iac or a dockerfile for an application that you know it's purl")
+
+	if err := producers.ParseFlags(); err != nil {
+		log.Fatal(err)
+	}
+
+	inFile, err := producers.ReadInFile()
+	if err != nil {
+		log.Fatal(err)
+	}
+	if err := run(inFile); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run(inFile []byte) error {
+	sarifResults, sarifErr := handleSarif(inFile)
+	cyclondxResults, cyclonedxErr := handleCycloneDX(inFile)
+	if sarifErr != nil && cyclonedxErr != nil {
```

Comment:
shouldn't you be checking if the sarif parsing worked and then skip the cyclonedx parsing if it did?